ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesfragment/valueobjects/FragmentInfo.java

```diff
@@ -46,8 +50,13 @@ public Boolean getImmutable() {
 		return immutable;
 	}
 
-	public void setImmutable(Boolean immutable) {
-		this.immutable = immutable;
+	public LocalDateTime getImmutableTimestamp() {
+		return immutableTimestamp;
+	}
+
+	public void makeImmutable() {
+		this.immutable = true;
```

Comment:
another possible solution would be to let isImmutable return wether immutableTimestamp is null. but I also approve this approach :)